parameters/examples/setup.rs

```diff
@@ -0,0 +1,304 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+use snarkvm_algorithms::{crh::sha256::sha256, CRH, SNARK, SRS};
+use snarkvm_dpc::{
+    InnerCircuit,
+    Network,
+    OuterCircuit,
+    PoSWScheme,
+    ProgramExecutable,
+    ProgramScheme,
+    SynthesizedCircuit,
+};
+use snarkvm_marlin::ahp::AHPForR1CS;
+use snarkvm_utilities::{FromBytes, ToBytes, ToMinimalBits};
+
+use anyhow::Result;
+use rand::{prelude::ThreadRng, thread_rng};
+use serde_json::{json, Value};
+use std::{
+    fs::File,
+    io::{BufWriter, Write},
+    path::PathBuf,
+};
+
+fn checksum(bytes: &[u8]) -> String {
+    hex::encode(sha256(bytes))
+}
+
+fn versioned_filename(filename: &str, checksum: &str) -> String {
+    match checksum.get(0..7) {
+        Some(sum) => format!("{}.{}", filename, sum),
+        _ => filename.to_string(),
+    }
+}
+
+/// Writes the given bytes to the given versioned filename.
+fn write_remote(filename: &str, version: &str, bytes: &[u8]) -> Result<()> {
+    let mut file = BufWriter::new(File::create(PathBuf::from(&versioned_filename(filename, version)))?);
+    file.write_all(&bytes)?;
+    drop(file);
```

Comment:
`file` will get dropped automatically at the end of the function (applies to the other functions here as well)